src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,456 @@
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_restart',
+        Str('name', required=True),
+        Bool('force'),
+    ))
+    @job(lock=GLUSTER_JOB_LOCK)
+    def restart(self, job, data):
+        """
+        Restart a gluster volume.
+
+        `name` Name of gluster volume
+        `force` Forcefully restart the gluster volume
+        """
+
+        name = data.pop('volname')
+
+        result = self.__volume_wrapper(volume.restart, name, **data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_stop',
+        Str('name', required=True),
+        Bool('force'),
+    ))
+    @job(lock=GLUSTER_JOB_LOCK)
+    def stop(self, job, data):
+        """
+        Stop a gluster volume.
+
+        `name` Name of gluster volume
+        `force` Forcefully stop the gluster volume
+        """
+
+        name = data.pop('name')
+
+        result = self.__volume_wrapper(volume.stop, name, **data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_delete',
+        Str('name', required=True),
+    ))
+    @job(lock=GLUSTER_JOB_LOCK)
+    def do_delete(self, job, data):
+        """
+        Delete a gluster volume.
+
+        `name` Name of the volume to be deleted
+        """
+
+        result = self.__volume_wrapper(volume.delete, data['name'])
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_info',
+        Str('name')
+    ))
+    @job(lock=GLUSTER_JOB_LOCK)
+    def info(self, job, data):
+        """
+        Return information about gluster volume(s).
+        If a gluster volume name has not been given,
+        this will return information about all volumes
+        detected in the cluster.
+
+        `name` Name of the gluster volume
+        """
+
+        name = data.get('name')
+
+        rv = {}
+        if name:
+            rv['volname'] = name
+
+        result = self.__volume_wrapper(volume.info, **rv)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_status',
+        Str('name')
+    ))
+    @job(lock=GLUSTER_JOB_LOCK)
+    def status(self, job, data):
+        """
+        Return detailed information about gluster volume(s).
+        If a gluster volume name has not been given,
+        this will return detailed information about all volumes
+        detected in the cluster.
+
+        `name` Name of the gluster volume
+        """
+
+        name = data.get('name')
+
+        rv = {}
+        if name:
+            rv['volname'] = name
+
+        result = self.__volume_wrapper(volume.status_detail, **rv)
+
+        return result
+
+    @accepts()
+    @job(lock=GLUSTER_JOB_LOCK)
+    def list(self, job):
+        """
+        Return list of gluster volumes.
+        """
+
+        result = self.__volume_wrapper(volume.vollist)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_optreset',
+        Str('name', required=True),
+        Str('opt'),
```

Comment:
Isnt opt required too? Or it will reset all options if not set?

Reply:
It will reset all options if not set.

Reply:
Its even written in the docstring, duh :P sorry